### DISCUSSION TITLE
Geographical search - what are the use cases?

### DISCUSSION THREAD

**Discussion by eklem** [fergiemcdowall/search-index#593] [Ideas]
Since search-inde v/3 we can sort a search result on what we want. With [latlon-geohash](https://github.com/chrisveness/latlon-geohash) this means we can do some nice things together with search-index. But what are the use cases?

Possible use cases
* Showing search results within a set of map tiles?
* Showing all hits within a set of map tiles?
* Sorting a search result on closeness to a point?
* Showing the closest other search results to one selected search result?

I was thinking of making a small library that could make it easy to use latlon-geohash together with search-index, but it would be nice to hear what people think are valid use cases.

A developer use case, I guess, is using it together with OpenstreetMap?

**Comment by eklem** [comment]
Grabbing some photo-data from Flickr, and generating a geohash for each photo. This will do as test-data I think.
https://github.com/eklem/geo-search-helper/blob/trunk/dataset/geophotos.json

**Comment by eklem** [comment]
Need to figure out distance between to points on a map (along a sphere suface), based on lon/lat:
https://www.calculator.net/distance-calculator.html?type=3&la1=59.917511442877185&lo1=10.753585322769617&la2=59.75006570875989&lo2=10.203666205253953&ctype=dec&lad1=38&lam1=53&las1=51.36&lau1=n&lod1=77&lom1=2&los1=11.76&lou1=w&lad2=39&lam2=56&las2=58.56&lau2=n&lod2=75&lom2=9&los2=1.08&lou2=w&x=67&y=20#latlog

**Comment by eklem** [comment]
First version of the [`geo-search-helper`](https://github.com/eklem/geo-search-helper) is ready. Will make a demo where you sort a search result on your position and can show the nearest results in a map.

And for use cases, I got some replies on Twitter, and it was mostly about finding stuff from where you were now. So, sorting a text search result on proximity of where you are now, and show all, some, within x km of where you are, in a map. And then do stuff like route from "here" to a search result.

So find your way to a result.

**Reply by fergiemcdowall** [reply]
Sounds interesting- looking forward to checking it out 👍🙂

**Reply by eklem** [reply]
I'll shout out when I have one more function in place and are able to show a "search" result in:
* a map, centered on where you are
* a listing, sorted on proximity on your position.

After that, I'm ready for a test with actual results from search-index. Figured out that you could do filtering on longitude/latitude too, with one of the helper functions.

**Reply by eklem** [reply]
I want to be able to find a map boundary for any given sets of graphical points. So, give the function i.e. the 10 closest search results and get back a map boundary (SW lon/lat + NE lon/lat) that will be able to show all the search results.
